packages/rocketchat-utils/server/lib/composeMessageObjectWithUser.js

```diff
@@ -1,21 +1,29 @@
 import { Users } from 'meteor/rocketchat:models';
 import { settings } from 'meteor/rocketchat:settings';
 
-const getUser = (userId) => Users.findOneById(userId);
+const getUserByUsername = (username) => Users.findOneByUsername(username, { fields: { name: 1 } });
+const getNameOfUser = (username) => {
+	const user = getUserByUsername(username);
+	return user ? user.name : undefined;
+};
 
 export const composeMessageObjectWithUser = function(message, userId) {
 	if (message) {
 		if (message.starred && Array.isArray(message.starred)) {
 			message.starred = message.starred.filter((star) => star._id === userId);
 		}
-		if (message.u && message.u._id && settings.get('UI_Use_Real_Name')) {
-			const user = getUser(message.u._id);
-			message.u.name = user && user.name;
+		if (message.u && message.u.name && !settings.get('UI_Use_Real_Name')) {
```

Comment:
why did you change to remove the field instead of adding it?

Reply:
In this case, because the real name is already saved in the `Messages` collection(`mentions` and `username`). So if you are not using `Real Names`, we remove the real name.

Reply:
the problem is that property is not updated (like if an user changes its name), that's why we always inject it before sending to client. I don't even know why we save it there in the first place.

Reply:
I changed to the old behavior.